test/gtest/ucs/test_datatype.cc

```diff
@@ -1200,5 +1204,76 @@ UCS_TEST_F(test_datatype, ptr_map) {
         delete i->second;
     }
 
-    ucs_ptr_map_destroy(&ptr_map);
+    UCS_PTR_MAP_DESTROY(test_st, &ptr_map);
+}
+
+UCS_PTR_MAP_DEFINE(test_mt, 1);
+
+UCS_MT_TEST_F(test_datatype, ptr_map_mt, 4)
+{
+    static ucs_init_once_t init_once = UCS_INIT_ONCE_INITIALIZER;
+    static pthread_mutex_t mutex     = PTHREAD_MUTEX_INITIALIZER;
+    static UCS_PTR_MAP_T(test_mt) ptr_map;
+    std_map_t std_map;
+    ucs_status_t status;
+
+    UCS_INIT_ONCE(&init_once) {
+        ASSERT_UCS_OK(UCS_PTR_MAP_INIT(test_mt, &ptr_map));
+    }
+
+    barrier();
+
+    const size_t N = 10 * UCS_KBYTE;
+
+    for (size_t i = 0; i < N; ++i) {
+        char *ptr     = new char;
```

Comment:
minor: can be `ptr = &s[i]` where `s` is `std::vector s(10 * UCS_KBYTE, 'a')`, for example